dex/src/main/scala/com/wavesplatform/dex/caches/RateCache.scala

```diff
@@ -37,7 +38,7 @@ object RateCache {
   def apply(db: DB): RateCache = new RateCache {
 
     private val rateDB  = RateDB(db)
-    private val rateMap = new ConcurrentHashMap[IssuedAsset, Double](rateDB.getAllRates.asJava)
+    private val rateMap = new ConcurrentHashMap[IssuedAsset, Double] unsafeTap (_.putAll(rateDB.getAllRates.asJava))
```

Comment:
I've looked at `ConcurrentHashMap.scala` and what I see:
```java
    /**
     * Creates a new map with the same mappings as the given map.
     *
     * @param m the map
     */
    public ConcurrentHashMap(Map<? extends K, ? extends V> m) {
        this.sizeCtl = DEFAULT_CAPACITY;
        putAll(m);
    }
```

It looks like you're doing the same thing, but in a more complex way. Isn't it? :)

Reply:
Done